resource/filter.go

```diff
@@ -167,3 +174,57 @@ func (f Filter) matches(r *Resource) bool {
 	}
 	return false
 }
+
+func (c *CreatedTime) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var v interface{}
+	if err := unmarshal(&v); err != nil {
+		return err
+	}
+	switch value := v.(type) {
+	case time.Time:
+		t, _ := v.(time.Time)
+		*c = CreatedTime{t}
+		return nil
+	case string:
+		d, err := time.ParseDuration(value)
+		if err == nil {
+			*c = CreatedTime{time.Now().UTC().Add(- d)}
+			return nil
+		}
+		var t time.Time
+		err = yaml.Unmarshal([]byte("!!timestamp " + value), &t)
```

Comment:
@jckuester One thing that I forgot to mention.  I was having a really bad time with the original timestamp parsing.  None of the formats shown in the readme were working.  Additionally, none of the supported YAML timestamp formats were working, because it would complain about not being able to convert `!!str` to a time.  This code was a "work-around" to force it to parse the value as a YAML timestamp.  Unfortunately, golang doesn't support all the possible formats of YAML timestamp.  This seemed better than before, but I'm open to anything that makes it clearer/better.

Reply:
Thanks for the info. Just to clarify: 

> I was having a really bad time with the original timestamp parsing.

With original timestamp, you mean `time.Time`? It caused trouble in your custom YAML parser, but worked fine in the standard parser before (https://github.com/cloudetc/awsweeper/blob/master/resource/filter.go#L65)?

Reply:
Not exactly.  On the commit prior to this PR, try adding this test to filter_test.go.
```
func Test_ParseFile(t *testing.T) {
	input := []byte(`aws_instance:
  - id: ^foo.*
    created:
      before: 2018-06-14
      after: 2018-10-28 12:28:39 +0000`)

 	var cfg resource.Config
	err := yaml.UnmarshalStrict(input, &cfg)
	require.NoError(t, err)
	require.NotNil(t, cfg[resource.Instance])
	require.Len(t, cfg[resource.Instance], 1)
	require.NotNil(t, cfg[resource.Instance][0].Created.Before)
	assert.Equal(t, time.Date(2018, 6, 14, 0, 0, 0, 0000, time.UTC), *cfg[resource.Instance][0].Created.Before)
	require.NotNil(t, cfg[resource.Instance][0].Created.After)
	assert.Equal(t, time.Date(2018, 10, 28, 12, 28, 39, 0000, time.UTC), *cfg[resource.Instance][0].Created.After)
}
```

When I run it I get this error:
```
$ make test
mkdir -p resource/mocks
go generate
go clean -testcache github.com/cloudetc/awsweeper github.com/cloudetc/awsweeper/command github.com/cloudetc/awsweeper/resource github.com/cloudetc/awsweeper/resource/mocks github.com/cloudetc/awsweeper/test
go test --race -cover github.com/cloudetc/awsweeper github.com/cloudetc/awsweeper/command github.com/cloudetc/awsweeper/resource github.com/cloudetc/awsweeper/resource/mocks github.com/cloudetc/awsweeper/test
?   	github.com/cloudetc/awsweeper	[no test files]
?   	github.com/cloudetc/awsweeper/command	[no test files]
--- FAIL: Test_ParseFile (0.00s)
        Error Trace:    filter_test.go:105
        Error:      	Received unexpected error:
                    	parsing time "2018-10-28 12:28:39 +0000" as "2006-01-02T15:04:05Z07:00": cannot parse " 12:28:39 +0000" as "T"
        Test:       	Test_ParseFile
FAIL
coverage: 33.6% of statements
FAIL	github.com/cloudetc/awsweeper/resource	0.030s
?   	github.com/cloudetc/awsweeper/resource/mocks	[no test files]
ok  	github.com/cloudetc/awsweeper/test	1.071s	coverage: 0.0% of statements
make: *** [test] Error 1
```